src/Compilers/Core/Portable/Compilation/IImportScope.cs

```diff
@@ -0,0 +1,111 @@
+    /// import directives.  There will also be an <see cref="IImportScope"/> for the containing compilation-unit if it
+    /// includes any import directives or if there are global import directives pulled in from other files.
+    /// </item>
+    /// <item>
+    /// In Visual Basic there will only be at most two <see cref="IImportScope"/>s returned for any position.  There
+    /// will be a scope for the containing compilation unit if it includes any import directives.  There can also be a
+    /// scope representing any imports specified at the project level.
+    /// </item>
+    /// <item>
+    /// Elements of any property have no defined order.  Even if they represent items from a single document, they are
+    /// not guaranteed to be returned in any specific file-oriented order.
+    /// </item>
+    /// <item>There is no guarantee that the same scope instances will be returned from successive calls to <see
+    /// cref="SemanticModel.GetImportScopes"/>.</item> 
+    /// </list>
+    /// </remarks>
+    public interface IImportScope
+    {
+        /// <summary>
+        /// Aliases defined at this level of the chain.  This corresponds to <c>using X = TypeOrNamespace;</c> in C# or
+        /// <c>Imports X = TypeOrNamespace</c> in Visual Basic.  This will include global aliases if present for both
+        /// languages.
+        /// </summary>
+        ImmutableArray<IAliasSymbol> Aliases { get; }
+
+        /// <summary>
+        /// Extern aliases defined at this level of the chain.  This corresponds to <c>extern alias X;</c> in C#.  It
+        /// will be empty in Visual Basic.
+        /// </summary>
+        ImmutableArray<IAliasSymbol> ExternAliases { get; }
+
+        /// <summary>
+        /// Types or namespaces imported at this level of the chain.  This corresponds to <c>using Namespace;</c> or
+        /// <c>using static Type;</c> in C#, or <c>Imports TypeOrNamespace</c> in Visual Basic.  This will include
+        /// global namespace or type imports for both languages.
+        /// </summary>
+        ImmutableArray<ImportedNamespaceOrType> Imports { get; }
+
+        /// <summary>
+        /// Xml namespaces imported at this level of the chain.  This corresponds to <c>Imports &lt;xmlns:prefix =
+        /// "name"&gt;</c> in Visual Basic.  It will be empty in C#.
+        /// </summary>
+        ImmutableArray<ImportedXmlNamespace> XmlNamespaces { get; }
+    }
+
+    /// <summary>
+    /// Represents an <see cref="INamespaceOrTypeSymbol"/> that has been imported, and the location the import was
+    /// declared at.  This corresponds to <c>using Namespace;</c> or <c>using static Type;</c> in C#, or <c>Imports
+    /// TypeOrNamespace</c> in Visual Basic.
+    /// </summary>
+    public readonly struct ImportedNamespaceOrType
+    {
+        public INamespaceOrTypeSymbol NamespaceOrType { get; }
+        public SyntaxReference? DeclaringSyntaxReference { get; }
```

Comment:
When would we not have syntax to reference? May be worth a comment. 
Also applies to `ImportedXmlNamespace.DeclaringSyntaxReference`

Reply:
will do.

Reply:
done.